### PR TITLE
Enable posts to have a list of authors

### PR DESCRIPTION
**Pull request by charlieegan3**
Sometimes posts have more than one author, it'd be nice to represent
this on post pages.

This reads from `author` or `authors` from the front matter to build a
list of authors. The list of authors is used to draw the page (list of
names at the top, list of 'profiles' at the bottom).

### CODE REVIEW

**Review by tomsabin — DISMISSED** [review]
Two things:
1. Could we change the base of this branch to staging?
2. We are missing the authors on the index page when there are multiple:
   localhost-4567-blog iphone 4

with author:
localhost-4567-blog iphone 4 1

I'm not sure if that was intentional but we do then have an empty div

**Comment by h-lame** [comment]
We've got the logic for finding the author spread across 4 templates now (we already did I think) but now it's more complex (author vs authors) so I think it's time to write a helper or something to extract the authors correctly.

**Comment by charlieegan3** [comment]
Thanks @h-lame and @tomsabin; I could pick this up later on but I've asked @cih to pick it up for the time being so this doesn't holdup their blog post.

File: lib/former_employee.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class FormerEmployee
```
Comment (h-lame) [inline lib/former_employee.rb]: Nice to see a null object here.  One problem is that not all authors who don't appear in the `data.people` are ex employees - we've had some guest bloggers before and maybe even clients?  So perhaps `NonEmployee` would be more correct (albeit not as snappy, better names are possible I'm sure).

File: source/blog/2016-10-28-next-london-2016-google-cloud-platform-s-user-conference.md [hunk 2]
```diff
@@ -2,9 +2,9 @@
 date: '2016-10-28 10:39 +0100'
 published: true
 title: NEXT London 2016 - Google Cloud Platform's User Conference
-authors: 
+author: Charlie Egan
+coauthors:
```
Comment (h-lame) [inline source/blog/2016-10-28-next-london-2016-google-cloud-platform-s-user-conference.md]: Why the split from author into author + coauthors ?  It implies a hierarchy of authorship that we might not intend (how to decide who is the author and who is the coauthor, especially meaningless as we don't distinguish in the page rendering).  I'd be in favour of saying we don't have author anymore, just authors (and do a grep to fix all the old frontmatter).

File: lib/former_employee.rb (hunk 1 above)
Comment (cih) [reply lib/former_employee.rb]: Ah yes I hadn't realised that, thanks I will update.

File: source/blog/2016-10-28-next-london-2016-google-cloud-platform-s-user-conference.md (hunk 2 above)
Comment (cih) [reply source/blog/2016-10-28-next-london-2016-google-cloud-platform-s-user-conference.md]: The reason I changed it from the initial author and authors is that I didn't want to have them being one or the other. But agree that if we remove the author and just go with authors consistently that will be good.

**Pull request merged**